common.rkt

```diff
@@ -165,24 +165,31 @@
   (reify initial-var st))
 
 (define (term<? u v)
+  (eqv? (term-compare u v) -1))
+
+(define (term-compare u v)
   (cond
-    ((null? u) (not (null? v)))
-    ((null? v) #f)
-    ((not u) (not v))
-    ((not v) #f)
-    ((eqv? u #t) v)
-    ((eqv? v #t) #f)
-    ((number? u) (or (not (number? v)) (< u v)))
-    ((number? v) #f)
-    ((symbol? u) (or (not (symbol? v)) (symbol<? u v)))
-    ((symbol? v) #f)
-    ((string? u) (or (not (string? v)) (string<? u v)))
-    ((string? v) #f)
-    ((pair? u) (or (not (pair? v)) (or (term<? (car u) (car v)) 
-                                       (and (equal? (cdr u) (cdr v)) 
-                                            (term<? (cdr u) (cdr v))))))
-    ((pair? v) #f)
-    (else #f)
+    ((eqv? u v) 0)
+    ((null? u) -1)
+    ((null? v) 1)
+    ((not u) -1)
+    ((not v) 1)
+    ((eqv? u #t) -1)
+    ((eqv? v #t) 1)
+    ((number? u) (if (number? v) (if (< u v) -1 1) -1))
+    ((number? v) 1)
+    ((symbol? u) (if (symbol? v) (if (symbol<? u v) -1 1) -1))
+    ((symbol? v) 1)
+    ((string? u) (if (string? v) (if (string<? u v) -1 1) -1))
+    ((string? v) 1)
+    ((pair? u) (if (pair? v) 
+                   (let ((compared-cars (term-compare (car u) (car v))))
+                     (if (eqv? compared-cars 0)
+                         (term-compare (cdr u) (cdr v))
+                         compared-cars))
+                   -1))
+    ((pair? v) 1)
+    (else 1)
```

Comment:
Put the closing parentheses on the same line.  Make sure this is the case elsewhere too.